### ISSUE TITLE
ViewMapAnnotation doesn't support clustering

### ISSUE THREAD

**Issue by bryan1anderson** [pauljohanneskraft/Map#27]
<img width="296" alt="Screenshot 2022-11-27 at 1 26 15 PM" src="https://user-images.githubusercontent.com/7976521/204157990-74afb0f8-b7c7-4c1f-b47d-1aa55c009816.png">
Unfortunately custom annotation views are required for my project. However, I've noticed that clustering gets turned off if I use ViewMapAnnotation

If I use MapMarker instead for example, it is clustered as anticipated 
<img width="267" alt="Screenshot 2022-11-27 at 1 27 30 PM" src="https://user-images.githubusercontent.com/7976521/204158043-98d3a76a-def9-4e01-b8d4-828eb632566b.png">

**Comment by pauljohanneskraft** [comment]
Yes, this is an issue of the current implementation. One would need to implement `mapView(_:clusterAnnotationForMemberAnnotations:)` in Map.Coordinator and implement it accordingly.

I'm not quite sure yet, how MapKit decides which member annotations to cluster and how we could reasonably provide the logic of creating the cluster without requiring another closure in the initializer or it only working for the built-in annotations (which probably the MapKit-SwiftUI approach is at the moment).

In your example, MapMarkers are created using MKMarkerAnnotationViews which probably have some built-in clustering mechanism even in MKMapView.

Would you like to further investigate or add your own thoughts about this? That would be highly appreciated, since I believe this is an important topic that should be approached soon, but I unfortunately do not yet have a full solution in mind.

**Comment by pauljohanneskraft** [comment]
Hey - I had a look at this and came up with this preliminary version. I haven't yet had the time to test the code thoroughly and am not entirely sure about the design, but maybe you have some feedback: #33 

The PR currently has some more changes that do not necessarily belong in there (e.g. the introduction of UserTrackingMode in favor of MKUserTrackingMode), but I haven't yet found the time to separate into its own PR.

**Comment by pauljohanneskraft** [comment]
I have just tried this with the following code and it seems to work perfectly fine:

```swift
import Map
import MapKit
import SwiftUI

struct Annotation: Identifiable {

    let id = UUID()
    let coordinate: CLLocationCoordinate2D
    let clusteringIdentifier: String

}

struct ContentView: View {

    @State private var region = MKCoordinateRegion(
        center: .init(latitude: 50, longitude: 10),
        latitudinalMeters: 100_000,
        longitudinalMeters: 100_000
    )

    @State private var annotations = [
        Annotation(coordinate: .init(latitude: 49.5, longitude: 09.5), clusteringIdentifier: "a"),
        Annotation(coordinate: .init(latitude: 49.5, longitude: 10.0), clusteringIdentifier: "a"),
        Annotation(coordinate: .init(latitude: 49.5, longitude: 10.5), clusteringIdentifier: "a"),
        Annotation(coordinate: .init(latitude: 50.0, longitude: 09.5), clusteringIdentifier: "b"),
        Annotation(coordinate: .init(latitude: 50.0, longitude: 10.0), clusteringIdentifier: "b"),
        Annotation(coordinate: .init(latitude: 50.0, longitude: 10.5), clusteringIdentifier: "b"),
        Annotation(coordinate: .init(latitude: 50.5, longitude: 09.5), clusteringIdentifier: "b"),
        Annotation(coordinate: .init(latitude: 50.5, longitude: 10.0), clusteringIdentifier: "b"),
        Annotation(coordinate: .init(latitude: 50.5, longitude: 10.5), clusteringIdentifier: "b"),
    ]


    var body: some View {
        Map(
            coordinateRegion: $region,
            annotationItems: annotations
        ) { annotation in
            ViewMapAnnotation(coordinate: annotation.coordinate, clusteringIdentifier: annotation.clusteringIdentifier) {
                Circle()
                    .fill((annotation.clusteringIdentifier == "a" ? Color.red : .yellow).opacity(0.5))
                    .frame(width: 16, height: 16)
            }
        } clusterAnnotation: { annotation, members in
            let _ = assert(Set(members.map(\.clusteringIdentifier)).count == 1)
            ViewMapAnnotation(annotation: annotation) {
                Circle()
                    .fill((members.first?.clusteringIdentifier == "a" ? Color.blue : .green).opacity(0.5))
                    .frame(width: 16, height: 16)
            }
        }
        .edgesIgnoringSafeArea(.all)
    }
}

```

**Comment by pauljohanneskraft** [comment]
Released as part of [Map 0.2.0](https://github.com/pauljohanneskraft/Map/releases/tag/0.2.0).
